labs/lab-10/tasks/overflow-for-binary/tests/test_overflow_for_binary.c

```diff
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <assert.h>
+#include <sys/wait.h>
+
+#include "graded_test.h"
+
+static int test(void)
+{
+	FILE *pipe = popen("python3 ../support/exploit.py", "r");
```

Comment:
When you use `popen()`, the working directory will be `tests/` so the `exploit.sh` script will error because it can't find `overflow_in_binary` in `tests/`. You'll find it easier to create some Bash checkers as stated in the issue [1] instead and just change the directory to `../support`.

[1] https://github.com/cs-pub-ro/operating-systems/tree/main/chapters/data/memory-security/drills/tasks/stack-protector/solution/tests